### PR TITLE
[0.x] Cleanup channels on removeSocket()

### PR DESCRIPTION
**Pull request by dkulyk**
Sometimes sockets still present in channels sets after disconnecting. 
I still could not find out why this happens.

### CODE REVIEW

File: src/namespace.ts [hunk 1]
```diff
@@ -39,8 +39,12 @@ export class Namespace {
     /**
      * Remove a socket from the namespace.
      */
-    removeSocket(wsId: string): Promise<boolean> {
-        return Promise.resolve(this.sockets.delete(wsId));
+    async removeSocket(wsId: string): Promise<boolean> {
+        this.channels.forEach(async (sockets, channel) => {
```
Comment (dkulyk) [inline src/namespace.ts:43]: forEach will not wait a promise resolving. It is like as without async/await

Comment (rennokki) [reply src/namespace.ts:43]: This can be better implemented with [async js](https://caolan.github.io/async/v3/docs.html#each)?

Comment (dkulyk) [reply src/namespace.ts:43]: I believe no sense to use async/await in current implementation

**Comment by rennokki** [comment]
This PR freaking solved the issues with most of the crazy assertions that used to fail in the past. You're a genius. 🎉

**Pull request merged**